ch_backup/logic/lock_manager.py

```diff
@@ -106,7 +108,23 @@ def _zk_flock(self) -> None:
             if not client.exists(self._process_zk_lockfile_path):
                 client.create(self._process_zk_lockfile_path, makepath=True)
             self._zk_lock = client.Lock(self._process_zk_lockfile_path)
-            if not self._zk_lock.acquire(blocking=False):
-                sys.exit(self._exitcode)
+            force_exit = self._exitcode == 0
+            try:
+                if not self._zk_lock.acquire(blocking=True, timeout=self._lock_timeout):
```

Comment:
As we are using `blocking=True`, `acquire()` raises an exception if the lock wasn't acquired within `timeout` seconds
https://github.com/python-zk/kazoo/blob/92bd0c2a8540697633564deed683f87d1f611bf5/kazoo/recipe/lock.py#L155
We shouldn't check the return value in that case.